modules/core/depinject.go

```diff
@@ -0,0 +1,74 @@
+package ibc
+
+import (
+	"cosmossdk.io/core/appmodule"
+	"cosmossdk.io/depinject"
+	storetypes "cosmossdk.io/store/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+
+	modulev1 "github.com/cosmos/ibc-go/api/ibc/core/module/v1"
+	capabilitykeeper "github.com/cosmos/ibc-go/modules/capability/keeper"
+	clienttypes "github.com/cosmos/ibc-go/v8/modules/core/02-client/types"
+	ibckeeper "github.com/cosmos/ibc-go/v8/modules/core/keeper"
+)
+
+var _ depinject.OnePerModuleType = AppModule{}
+
+// IsOnePerModuleType implements the depinject.OnePerModuleType interface.
+func (AppModule) IsOnePerModuleType() {}
+
+func init() {
+	appmodule.Register(
+		&modulev1.Module{},
+		appmodule.Provide(ProvideModule),
+	)
```

Comment:
I'm thinking that we may need something like `InvokeAddRoutes` here which will register `[]porttypes.IBCModule` on the router..

Something similar to x/gov maybe: https://github.com/cosmos/cosmos-sdk/blob/main/x/gov/depinject.go#L100

Hopefully we can use this as some kind of example. Happy to work on this once the initial PRs are merged.